### PR TITLE
pin compiler toolchain to version 10 on osx

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

**Comment by esc** [comment]
BuildFarmID: `llvmlite_112`

File: conda-recipes/llvmlite/conda_build_config.yaml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
   - 10                      # [osx]
 
-cxx_compiler_version:       # [linux and osx]
+cxx_compiler_version:
   - 7                       # [linux and (x86_64 or ppc64le)]
   - 9                       # [linux and aarch64]
   - 10                      # [osx]
 
-fortran_compiler_version:   # [linux]
+fortran_compiler_version:
```
Comment (katietz) [inline conda-recipes/llvmlite/conda_build_config.yaml:12]: you should make it again being linux specific

File: conda-recipes/llvmlite/conda_build_config.yaml [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
+c_compiler_version:
   - 7                       # [linux and (x86_64 or ppc64le)]
   - 9                       # [linux and aarch64]
   - 10                      # [osx]
 
-cxx_compiler_version:       # [linux and osx]
+cxx_compiler_version:
   - 7                       # [linux and (x86_64 or ppc64le)]
   - 9                       # [linux and aarch64]
   - 10                      # [osx]
```
Comment (katietz) [inline conda-recipes/llvmlite/conda_build_config.yaml:10]: you should use here instead (osx and x86_64), as for osx-arm64 there is no clang 10

File: conda-recipes/llvmlite/conda_build_config.yaml [hunk 3]
```diff
@@ -1,14 +1,14 @@
 # Numba/llvmlite stack needs an older compiler for backwards compatability.
-c_compiler_version:         # [linux and osx]
+c_compiler_version:
```
Comment (katietz) [inline conda-recipes/llvmlite/conda_build_config.yaml:2]: same here, use 'linux or (osx and x86_64)' ... the old line seems buggy anyway.  It should have been 'linux or osx'

File: conda-recipes/llvmlite/conda_build_config.yaml [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
   - 7                       # [linux and (x86_64 or ppc64le)]
   - 9                       # [linux and aarch64]
   - 10                      # [osx]
```
Comment (katietz) [inline conda-recipes/llvmlite/conda_build_config.yaml:5]: see 'osx and x86_64' ... from above

**Pull request merged**